site/src/pages/components/switch.explainer.mdx

```diff
@@ -0,0 +1,438 @@
+| `autofocus`   | `autofocus`    | `boolean` | `false`       | Get focus by default.                                                                                                                                                                                                          |
+| `required`    | `required`     | `boolean` | `false`       | Indicates that the switch is invalid unless on.                                                                                                                                                                                |
+| `form`        | `form`         | `string`  | `""`          | Associates the element with a form in the document whose `id` is this value.                                                                                                                                                   |
+
+#### Methods
+
+<div class="question">
+  The switch state can be toggled by setting `mySwitch.value = true`. Is a dedicated method
+  `mySwitch.toggle(): (force?: boolean) => void` necessary or helpful?
+</div>
+
+#### Events
+
+| Event Name | Detail Type | Bubbles | Composed | Cancellable | Dispatch Behavior                                      |
+| ---------- | ----------- | ------- | -------- | ----------- | ------------------------------------------------------ |
+| `change`   | none        | `true`  | `true`   | `false`     | Fired when the switch `value` changes.                 |
+| `input`    | none        | `true`  | `true`   | `false`     | Fired when the switch `value` is commited by the user. |
+
+### Appearance
+
+_Screenshots and/or description of the basic appearance of the component._
+
+### Anatomy
+
+{/* _Outline the component's structure with a diagram of its render tree. Enumerate key areas of visual composition and customization._ */}
+
+#### Switch with moving track
+
+<div class="question">
+  If the track content on either side can be of arbitrary length, the track width needs to be `2 *
+  longest side` and the switch needs to be `track width / 2` (not accounting for thumb width,
+  paddings and borders). Is it possible to style this?
+</div>
+
+Animation variant where the thumb and the track move together to reveal the hidden side of the track. Any text displayed on the switch is optional and defined by the author through a slot or other means.
+
+![Switch anatomy with moving track](../images/switch-moving-track-anatomy.png)
+_Source: [Figma source of switch anatomy with moving track](https://www.figma.com/file/uAYj6yVcAGD2OgjwCTqRol/Switch?type=design&node-id=2%3A28&mode=design&t=OAr7EJBXAQ4AEwfI-1)_
+
+Limitations:
+
+- The thumb needs to be at least the height of the track in order to hide the boundary between active and inactive track side underneath.
```

Comment:
I don't think this should be defined; why is a requirement/limitation? This is just a block within another block. A developer should be able to make them smaller, larger, etc than the track. The parts existence; is irrelevant to its visual capabilities. I'll note I may be mis-understanding your point here.

Reply:
I changed the section title to "edge cases" and updated the sample images to point out a possible implementation issue, which is closer to what I initially wanted to point out here.